lib/git_worktree.rb

```diff
@@ -139,7 +160,7 @@ def mv_dir(old_dir, new_dir)
 
   def fetch_and_merge
     fetch
-    commit = @repo.ref("refs/remotes/#{@remote_name}/master").target
+    commit = @repo.ref("refs/remotes/#{@remote_name}/#{@current_branch}").target
```

Comment:
But it makes the API weird...

```ruby
tree = GitWorktree.new(:commit_sha => "SOME_NONMASTER_SHA")
tree.fetch_and_merge # fetches master?
```

Perhaps what you are trying to accomplish is to specify the merge_target_branch (or merge_target_ref)?  @bdunne You just did something similar for the bot...thoughts?